net.c

```diff
@@ -2349,6 +2350,8 @@ void update_sysconfig(slist_t *slist, char *filename, char *filename_tmp) {
     else {
       log_info("warning: %s not updated\n", filename);
     }
+
+    free(filename_tmp);
```

Comment:
asprintf() may leave filename_tmp undefined on error.

C is a pain sometimes. :-)